### ISSUE TITLE
Alchemy shouldn't let you Redeem if the DAO has no funds

### ISSUE THREAD

**Issue by stratisdaostack** [daostack/alchemy#934]
<img width="522" alt="Screenshot 2019-07-18 at 09 48 04" src="https://user-images.githubusercontent.com/41568024/61435601-86bccf80-a941-11e9-876c-c9e33f3a6a8b.png">

This is from PolkaDAO which has no GEN holdings but I was still able to click Redeem and of course **pay the TX fee.**

**Comment by adrien-be** [comment]
Good point Stratis. 
Yes, client level is definitely enough.

Note that it is still not clear whether one should be able to redeem part of the rewards, for instance: the DAO has GEN but no DAI, you are eligible for GEN & DAI, if clicking "redeem" then what should be the behavior? a/ not being able to redeem anything    b/ get the GEN and redeem again later (can be weeks) when the DAO has DAI

**Comment by jellegerbrandy** [comment]
suppose you are eligibable for gen and dai, the dao has gen but no dai:
- when click redeem, you get your gen
- the proposal stays among redeemable proposals, bc there is still some dai to redeem (even if the dao cannot pay you). If you click "redeem", you'll get a warning ("the dao has insufficient funds"), and the transaction will not be sent.

**Comment by adrien-be** [comment]
Thanks. I've noticed that behavior afterwards. I've discussed the pros & cons of that implementation with Eylon (or Alex? I'm not sure) some time ago.
Pros: 
 a/ it makes sure one can get the available assets, which in some case is key to going forward with a proposal
Cons: 
 a/ it makes bookkeeping a more complex
 b/ it makes the implementation more complex (my guess is that you will have more bugs as a result)

Just some thoughts, no strong opinion on that specific topic.
